packages/salesforce-adapter/src/filters/elements_url.ts

```diff
@@ -0,0 +1,81 @@
+/*
+*                      Copyright 2020 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { logger } from '@salto-io/logging'
+import { BuiltinTypes, CORE_ANNOTATIONS, Element, ElementIDResolver, isObjectType } from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { resolvePath } from '@salto-io/adapter-utils'
+import wu from 'wu'
+import { promises } from '@salto-io/lowerdash'
+import { FilterCreator } from '../filter'
+import { lightiningElementsUrlRetreiver } from '../elements_url_retreiver/elements_url_retreiver'
+
+const MAX_UPDATES_ELEMENT_URL_CONCURRENCY = 100
+
+const log = logger(module)
+
+const createElementIdResolver = (elementsMap: Record<string, Element>): ElementIDResolver =>
+  async elementID => {
+    const { parent } = elementID.createTopLevelParentID()
+    const parentElement = elementsMap[parent.getFullName()]
+    return resolvePath(parentElement, elementID)
+  }
+
+
+function *getRelevantElements(elements: Element[]): IterableIterator<Element> {
+  for (const element of elements) {
+    yield element
+    if (isObjectType(element)) {
+      for (const field of Object.values(element.fields)) {
+        yield field
+      }
+    }
+  }
+}
```

Comment:
we don't need to write loops, this is equivalent to
```suggestion
wu(elements)
  .map(elem => isCustomObject(elem) ? [e, ...Object.values(elem.fields)] : [e])
  .flatten(true)
```
which is both shorter and easier to read (once you get used to our code style), and a repeating pattern in our code (search for `Object.values(.*.fields)` in the salesforce adapter package)
also, using `isCustomObject` instead of `isObjectType` can save us some iterations because we know only custom object fields are relevant